sdk/include/opentelemetry/sdk/common/http_client.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/function_ref.h"
+#include "opentelemetry/version.h"
+
+/*
+ Usage Example
+
+ struct SimpleReponseHandler: public ResponseHandler {
+      void OnResponse(Response& res) noexcept override
+      {
+           if (res.IsSuccess()) {
+            res.GetNextHeader([](std::string name, std::string value) -> bool {
+                std::cout << "Header Name:" << name << " Header Value:"<< value ;
+                return true;
+            });
+            .. process response body
+           }
+      }
+  };
+
+  SessionManager sessionManager; // implementer can provide singleton implementation for it
+  auto session = sessionManager.createSession("localhost", 8000);
+  auto request = session->CreateRequest();
+  request->AddHeader(..);
+  SimpleResponseHandler res_handler;
+  session->SendRequest(res_handler);
+  session->FinishSession() // optionally in the end
+  ...shutdown
+  sessionManager.FinishAllSessions()
+*/
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+namespace http
+{
+
+enum class Method
+{
+  Get,
+  Post,
+  Put,
+  Options,
+  Head,
+  Patch,
+  Delete
+};
+
+enum class SessionState
+{
+  Created,  // session object is created
+  Ongoing,  // http(s) request is ongoing
+  Finished  // https(s) request is finished
+};
+
+using Body       = std::vector<uint8_t>;
+using StatusCode = uint16_t;
+
+class Request
+{
+
+public:
+  virtual void setMethod(Method method) noexcept = 0;
+
+  virtual void setUri(const std::string &uri) noexcept = 0;
+
+  virtual void SetBody(Body &body) noexcept = 0;
+
+  virtual void AddHeader(const std::string &name, const std::string &value) noexcept = 0;
+
+  virtual void ReplaceHeader(const std::string &name, const std::string &value) noexcept = 0;
```

Comment:
Not critical: we could replace all `const std::string &` with `nostd::string_view`. This will avoid construction of intermediate strings when people pass in `const char *`.

Reply:
done.